src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -1555,6 +1539,164 @@ matchPointsToValue opts sc cc spec prepost loc maybe_cond ptr val =
+     let memTy = biBitfieldType bfIndex
+     storTy <- Crucible.toStorableType memTy
+     res <- liftIO $ Crucible.loadRaw sym mem ptr storTy alignment
+     let badLoadSummary = summarizeBadLoad cc memTy prepost ptr
+     case res of
+       Crucible.NoErr pred_ res_val -> do
+         addAssert pred_ $ Crucible.SimError loc $
+           Crucible.AssertFailureSimError (show $ PP.vcat badLoadSummary) ""
+         case res_val of
+           -- This will only work if:
+           --
+           -- * The bitfield is in fact a bitvector, and
+           -- * The width of the RHS type is less than the width of the
+           --   bitfield type.
+           --
+           -- We check these criteria in this case.
+           Crucible.LLVMValInt bfBlk bfBV
+             |  let bfWidth = W4.bvWidth bfBV
+             ,  Some rhsWidth <- mkNatRepr $ fromIntegral $ biFieldSize bfIndex
+             -> case (testLeq (knownNat @1) rhsWidth, testLeq (incNat rhsWidth) bfWidth) of
+                  (Just LeqProof, Just LeqProof) ->
+                    do -- Here is where we perform the bit-twiddling needed
+                       -- to match the RHS value against the field within the
+                       -- bitfield. We will use this as a running example:
+                       --
+                       --   struct s {
+                       --     int32_t w;
+                       --     uint8_t x1:1;
+                       --     uint8_t x2:2;
+                       --     uint8_t y:1;
+                       --     int32_t z;
+                       --   };
+                       --
+                       -- Let us imagine that we are matching against the
+                       -- `y` field. The bitfield (bfBV) will look something
+                       -- like 0b????Y???, where `Y` denotes the value of the
+                       -- `y` bit.
+                       let -- The offset (in bits) of the field within the
+                           -- bitfield. For `y`, this is 3 (x1's offset is 0
+                           -- and `x2`'s offset is 1).
+                           bfOffset = biFieldOffset bfIndex
+                           bfOffsetBV = BV.mkBV bfWidth $ fromIntegral bfOffset
+
+                       -- Use a logical right shift to make the bits of the
+                       -- field within the bitfield become the least
+                       -- significant bits. In the `y` example, this is
+                       -- tantamount to shifting 0b????Y??? right by `bfOffset`
+                       -- (i.e., 3 bits) to become 0b000????Y.
+                       bfBV'  <- liftIO $ W4.bvLshr sym bfBV =<< W4.bvLit sym bfWidth bfOffsetBV
+
+                       -- Finally, truncate the bitfield such that only the
+                       -- bits for the field remain. In the `y` example, this
+                       -- means truncating 0b000????Y to `0bY`.
+                       bfBV'' <- liftIO $ W4.bvTrunc sym rhsWidth bfBV'
```

Comment:
this can be `W4.bvSelect sym bfOffset_repr rhsWidth bfBV`.

Reply:
Ah, good suggestion. I've submitted #1547 to implement this.